### PR TITLE
fix: show username on cert if name is private

### LINKED ISSUE #40765: Private names show up on certificates
If I set my name to private on my settings page, it still shows up on certificates.

We could perhaps do like we do for some of the other settings and disable certificates when a name is private.

Or perhaps just use a generic or blank name on the certificate - this doesn't sound very good though - I vote for the first option. It's not that big of a problem, but it seems like we shouldn't show a users name anywhere if it's set to private.

### PR DESCRIPTION
**Pull request by moT01**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #40765

### CODE REVIEW

**Comment by ShaunSHamilton** [comment]
Ideally, should we be **not** even fetching `cert.name`, if `showName` is false?

**Comment by moT01** [comment]
Good call @ShaunSHamilton 👍 - I think I fixed it.

File: client/src/client-only-routes/ShowCertification.js [hunk 1]
```diff
@@ -230,13 +233,16 @@ const ShowCertification = props => {
...
-    name: userFullName,
+    name: userFullName = null,
     username,
     certTitle,
     completionTime
   } = cert;
+
   const { user } = props;
 
+  const displayName = userFullName ? userFullName : username;
```
Comment (ShaunSHamilton) [inline client/src/client-only-routes/ShowCertification.js:244]:
```suggestion
  const displayName = userFullName ?? username;
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Small change, just because I like this wherever I can.

**Review by raisedadead — APPROVED** [review]

**Pull request merged**